[Bug] FAISS 向量索引维度未适配配置中的 embedding_dimensions: 0，导致 qwen3-embedding-8b 报错 AssertionError

### What happened / 发生了什么

在配置中使用 `embedding_dimensions: 0`（自动模式）时，FAISS 向量索引未能正确获取嵌入模型的实际维度（应为 4096），而是使用了其他维度创建索引，导致向量检索时抛出 `AssertionError: assert d == self.d`。

### Reproduce / 如何复现？

1. 在 `cmd_config.json` 中配置嵌入模型 `qwen3-embedding-8b`，并设置 `"embedding_dimensions": 0` 和 `"embedding_dimensions_mode": "auto"`。
2. 安装并启用 `livingmemory` 插件。
3. 向机器人发送任意消息，触发记忆召回。

### AstrBot version, deployment method (e.g., Windows Docker Desktop deployment), provider used, and messaging platform used. / AstrBot 版本、部署方式（如 Windows Docker Desktop 部署）、使用的提供商、使用的消息平台适配器

AstrBot 版本: v4.26.7

部署方式: Docker

使用的提供商: ggwk1（OpenAI 兼容 API）

嵌入模型: qwen3-embedding-8b

消息平台适配器: aiocqhttp (Napcat)

操作系统: Linux

### OS

Linux

### Logs / 报错日志

[2026-07-25 00:52:55.921] [Plug] [INFO] [core.plugin_initializer:683]: LivingMemory 插件初始化成功。
[2026-07-25 00:54:12.540] [Core] [INFO] [aiocqhttp.aiocqhttp_platform_adapter:107]: aiocqhttp(OneBot v11) 适配器已连接。
[2026-07-25 00:54:27.348] [Core] [INFO] [core.event_bus:74]: [default] [huli(aiocqhttp)] 用户: 你好 
[2026-07-25 00:54:27.556] [Plug] [INFO] [event_handler_modules.memory_recall:90]: 已归一化 1 条纯文本历史消息
[2026-07-25 00:54:27.570] [Plug] [INFO] [utils.__init__:222]: [get_persona_id] 最终使用人格: 狐狸SFW
[2026-07-25 00:54:27.571] [Plug] [INFO] [event_handler_modules.memory_recall:191]: 开始记忆召回，查询='你好...'
[2026-07-25 00:54:27.573] [Plug] [INFO] [managers.memory_engine:2059]: [自动迁移] 开始检查会话，候选匹配: ['用户ID']
[2026-07-25 00:54:29.056] [Plug] [ERRO] [v4.26.7] [utils.__init__:167]: 操作失败: 记忆召回 (耗时 1.500s) - 
[2026-07-25 00:54:29.059] [Plug] [ERRO] [v4.26.7] [event_handler_modules.memory_recall:294]: 处理 on_llm_request 钩子时发生错误: 
Traceback (most recent call last):
  File "/AstrBot/data/plugins/astrbot_plugin_livingmemory/core/event_handler_modules/memory_recall.py", line 195, in handle_memory_recall
    recalled_memories = await self.memory_engine.search_memories(
  File "/AstrBot/data/plugins/astrbot_plugin_livingmemory/core/managers/memory_engine.py", line 1189, in search_memories
    results = await self.dual_route_retriever.search(
  File "/AstrBot/data/plugins/astrbot_plugin_livingmemory/core/retrieval/dual_route_retriever.py", line 44, in search
    doc_results, graph_results = await asyncio.gather(
  File "/AstrBot/data/plugins/astrbot_plugin_livingmemory/core/retrieval/graph_retriever.py", line 64, in search
    keyword_results, vector_results = await asyncio.gather(
  File "/AstrBot/data/plugins/astrbot_plugin_livingmemory/core/retrieval/graph_vector_retriever.py", line 60, in search
    raw_results = await self.faiss_db.retrieve(
  File "/AstrBot/astrbot/core/db/vec_db/faiss_impl/vec_db.py", line 221, in retrieve
    scores, indices = await self.embedding_storage.search(
  File "/AstrBot/astrbot/core/db/vec_db/faiss_impl/embedding_storage.py", line 72, in search
    distances, indices = self.index.search(vector, k)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/faiss/class_wrappers.py", line 437, in replacement_search
    assert d == self.d
           ^^^^^^^^^^^
AssertionError: assert d == self.d

### 报错分析

查询向量维度为 4096，但 FAISS 索引使用了其他维度（推测为 1024）。

### 配置信息

```json
"embedding_dimensions": 0,
"embedding_dimensions_mode": "auto"

### Are you willing to submit a PR? / 你愿意提交 PR 吗？

- [ ] Yes!

### Code of Conduct

- [x] I have read and agree to abide by the project's [Code of Conduct](https://docs.github.com/zh/site-policy/github-terms/github-community-code-of-conduct)。

--- a/astrbot/core/db/vec_db/faiss_impl/embedding_storage.py
+++ b/astrbot/core/db/vec_db/faiss_impl/embedding_storage.py
@@ -73,6 +73,11 @@ class EmbeddingStorage:
         if path and os.path.exists(path):
             self.index = self._read_index(path)
         else:
+            if dimension <= 0:
+                raise ValueError(
+                    f"无效的嵌入向量维度: {dimension}。请检查该知识库使用的 Embedding "
+                    "Provider 是否正确配置了 embedding_dimensions。",
+                )
             base_index = faiss.IndexFlatL2(dimension)
             self.index = faiss.IndexIDMap(base_index)
 

--- a/tests/unit/test_faiss_vec_db.py
+++ b/tests/unit/test_faiss_vec_db.py
@@ -3,6 +3,7 @@ from unittest.mock import AsyncMock
 
 import pytest
 
+from astrbot.core.db.vec_db.faiss_impl.embedding_storage import EmbeddingStorage
 from astrbot.core.db.vec_db.faiss_impl.vec_db import FaissVecDB
 from astrbot.core.exceptions import KnowledgeBaseUploadError
 from astrbot.core.provider.provider import EmbeddingProvider
@@ -62,6 +63,22 @@ async def test_insert_batch_raises_friendly_error_for_embedding_count_mismatch()
     assert "期望 2，实际 1" in str(exc_info.value)
     vec_db.document_storage.insert_documents_batch.assert_not_awaited()
     vec_db.embedding_storage.insert_batch.assert_not_awaited()
+
+
+def test_embedding_storage_rejects_zero_dimension_for_a_fresh_index(tmp_path) -> None:
+    with pytest.raises(ValueError, match="无效的嵌入向量维度"):
+        EmbeddingStorage(0, str(tmp_path / "index.faiss"))
+
+
+def test_embedding_storage_rejects_negative_dimension_for_a_fresh_index() -> None:
+    with pytest.raises(ValueError, match="无效的嵌入向量维度"):
+        EmbeddingStorage(-1)
+
+
+def test_embedding_storage_accepts_a_valid_dimension_for_a_fresh_index() -> None:
+    storage = EmbeddingStorage(4)
+
+    assert storage.index.d == 4
 
 
 @pytest.mark.asyncio

